### PR TITLE
Update plan_id label in CR

### PR DESCRIPTION
**Pull request by visargsoneji**
The plain_id was not getting updated in case of updateInstance call. This PR fixes that issue!

### CODE REVIEW

**Review by jintusebastian — APPROVED** [review]

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n+n+n failed.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n is succeeded.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n+n+n failed.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n-1 is succeeded.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n+n+n failed.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n-2 is succeeded.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n+n+n failed.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n-1 is succeeded.

**Comment by vinaybheri** [comment]
Validation on k8s cluster: n-2 is succeeded.

**Review by anoopjb — APPROVED** [review]

File: broker/applications/osb-broker/src/api-controllers/ServiceBrokerApiController.js [hunk 1]
```diff
@@ -202,7 +202,10 @@ class ServiceBrokerApiController extends FabrikBaseController {
       res.status(statusCode).send(body);
     }
 
-    const labels = req.params.region ? { 'region':req.params.region } : undefined;
+    let labels = req.params.region ? { 'region':req.params.region } : undefined;
+    labels = _.merge(labels, {
```
Comment (anoopjb) [inline broker/applications/osb-broker/src/api-controllers/ServiceBrokerApiController.js:206]: Here were are updating the label whenever we gets a plan update request. But I feel there can be a case when the user sends a wrong plan to update to and our `isUpdatePossible` assertion rejects it. Then we may have a wrong plan id in the label.

**Review by anoopjb — CHANGES_REQUESTED** [review]

**Review by anoopjb — APPROVED** [review]

File: broker/applications/osb-broker/src/api-controllers/ServiceBrokerApiController.js (hunk 1 above)
Comment (anoopjb) [reply broker/applications/osb-broker/src/api-controllers/ServiceBrokerApiController.js:206]: Sorry, I realized that we are not applying the label is there are errors. 👍

**Pull request merged**